src/lib/daemon_rpcs/daemon_rpcs.ml

```diff
@@ -318,166 +327,420 @@ module Types = struct
 end
 
 module Send_user_command = struct
-  type query = User_command.Stable.Latest.t [@@deriving bin_io]
+  module Query = struct
+    [%%versioned
+    module Stable = struct
+      module V1 = struct
+        type t = User_command.Stable.V1.t
 
-  type response = Receipt.Chain_hash.Stable.Latest.t Or_error.t
-  [@@deriving bin_io]
+        let to_latest = Fn.id
+      end
+    end]
 
-  type error = unit [@@deriving bin_io]
+    type t = Stable.Latest.t
+  end
 
-  let rpc : (query, response) Rpc.Rpc.t =
-    Rpc.Rpc.create ~name:"Send_user_command" ~version:0 ~bin_query
-      ~bin_response
+  module Response = struct
+    [%%versioned
+    module Stable = struct
+      module V1 = struct
+        type t =
+          Receipt.Chain_hash.Stable.V1.t Core_kernel.Or_error.Stable.V1.t
+
+        let to_latest = Fn.id
+      end
+    end]
+
+    type t = Stable.Latest.t
+  end
+
+  let rpc : (Query.t, Response.t) Rpc.Rpc.t =
```

Comment:
I think think some of these RPC calls should be a certain version, not the latest version, no? We want to make sure a client is processing a specified version of an RPC right?

Reply:
The client and daemon are running the same version of the software.  

For the gossip network, the nodes can be running different versions of the software. There, we use the Jane St versioned-RPC calls so that nodes can negotiate versions.

We haven't done that for the client-daemon interactions.